### ISSUE TITLE
Can tab outside of a callout

### ISSUE THREAD

**Issue by betrue-final-final** [OfficeDev/office-ui-fabric-react#1597]
Environment: IE11 and JAWS 18
Are you willing to submit a PR to fix? No. I need help
Requested priority: High

1. When invoked, callout does not announce itself.
Expected: Callout says "<callout name>, alert, <description>"

2. Can tab out of callout. Focus should be trapped.

**Comment by mdahamiwal** [comment]
> When invoked, callout does not announce itself.
Expected: Callout says ", alert, "

Don't see a reason why a callout need to be an alert as far as we move the focus to it, moreover, we use callout in lot other components like dropdown, tooltip, contextmeu. Adding alert to callout will be a bad experience.

> 2. Can tab out of callout. Focus should be trapped.

Is there any guideline that specifies this behavior? We generally follow one of the aria design pattern, given that callout is a multipurpose container, its content should decide which pattern to follow. Thoughts ?

**Comment by betrue-final-final** [comment]
We need to tell the user that they are in a new object on the page if we bring focus to it. I've been listening to  Windows toasts, and they don't grab focus, so we could announce them without moving focus.

As for the second issue, one of the big problems is that if you tab, you end up at the top of the tab order, in the browser chrome. This is totally disorienting. You are in a new surface that has no relationship to the main page. It's like a dialog, so escape should  be the way to get out of it, and bring you back to where you were before the interruption.

**Comment by AbhinavSinghT** [comment]
In relation to issue #2, tab order feels broken when tabbing out of the callout. Should trap the tab or provide a way to navigate to next control and not to the top tab stop. Narration is painful when callout loses control due to tab.

**Comment by betrue-final-final** [comment]
Yes, I agree @abhinavsinght. We need to trap focus.

**Comment by betrue-final-final** [comment]
This is still an issue, for both focustrap and calling itself out.

**Comment by betrue-final-final** [comment]
All previous issues are fixed except for the focuszone issue.

**Comment by yumikohey** [comment]
@betrue-final-final The workaround that I tried is added a `FocusTrapZone` within `CallOut`.

**Comment by micahgodbolt** [comment]
Assigning shield dev to take a look.

**Comment by atneik** [comment]
I propose having a prop, lets say `trapFocus`, when set to `true` adds the `FocusTrapZone` and traps the focus. By default it will be `false` so that its not a breaking change. Having the prop should be ideal as @mdahamiwal mentioned its a multipurpose container and should support both behaviors.

**Comment by joschect** [comment]
If we do things that we the switch to using it is easy and will work exactly the same as a regular callout, but you get the benefit of the focustrapzone.

**Comment by atneik** [comment]
@joschect makes sense. I'll create a PR for to add above mentioned composite component

**Comment by tonytang-microsoft-com** [comment]
We have a button that shows a callout when clicked. We used FocusTrapCallout in order to fix this tab issue, however now the callout will only been dismissed when pressing ESC. FocusTrapCallout's onDismiss property is not getting called if I click out side of the callout. Is that a bug?

**Comment by kwilkins** [comment]
Seeing the same issue @tonytang-microsoft-com brought up. Is this expected? It's reproducible on the [Callout documentation page](https://developer.microsoft.com/en-us/fabric#/controls/web/callout)

**Comment by joschect** [comment]
@tonytang-microsoft-com yes, I believe that this is expected if `isClickableOutsideFocusTrap` is not set to true. If that's set to false then the focustrapzone will prevent clicks from being registered outside of itself.
